docs/src/06-contributing/01-contributing-guide.md

```diff
@@ -1 +1,261 @@
+(currently 3.5.6) and expose it via `SPARK_HOME`:
+```bash
+curl -O https://downloads.apache.org/spark/spark-3.5.6/spark-3.5.6-bin-hadoop3.tgz
+mkdir -p "$HOME/.local/spark"
+tar -xzf spark-3.5.6-bin-hadoop3.tgz -C "$HOME/.local/spark"
+export SPARK_HOME="$HOME/.local/spark/spark-3.5.6-bin-hadoop3"
+```
+Persist the environment variables in your shell profile:
+```bash
+export SPARK_HOME="$HOME/.local/spark/spark-3.5.6-bin-hadoop3"
+export PATH="$SPARK_HOME/bin:$PATH"
+```
+
+> **Tip:** To test against Spark 4.x, pass `-Dspark.version=<version>` when invoking `./build/sbt`.
+
+---
+
+## 2. Clone the repository
+
+1. Fork the repository on GitHub (recommended) and copy the clone URL.
+2. Clone your fork and switch into the workspace:
+   ```bash
+   git clone https://github.com/<your-user>/graphframes.git
+   cd graphframes
+   ```
+3. Configure the upstream remote to stay in sync:
+   ```bash
+   git remote add upstream https://github.com/graphframes/graphframes.git
+   git fetch upstream
+   ```
+4. Create a topic branch for your change:
+   ```bash
+   git checkout -b feature/my-change
+   ```
+
+---
+
+## 3. Scala / JVM workflow
+
+GraphFrames provides a checked-in sbt launcher at `./build/sbt`; a separate sbt
+installation is not required.
+
+### 3.1 Compile the project
+```bash
+./build/sbt compile
+```
+The first run downloads all dependencies and may take several minutes. If
+compilation fails with `-Xfatal-warnings` complaining about deprecated
+`java.net.URL` constructors, ensure the build is using your Java 17 toolchain. You
+can force it with:
+```bash
+./build/sbt -java-home "$JAVA_HOME" compile
+```
+
+### 3.2 Format Scala code
+```bash
+./build/sbt scalafmtAll test:scalafmt
+```
+
+### 3.3 Run Scala tests
+Run the full test suite:
+```bash
+./build/sbt test
+```
+Focus on a specific suite while iterating:
+```bash
+./build/sbt "core/testOnly org.graphframes.lib.PageRankSuite"
+```
+
+### 3.4 Helpful sbt tasks
+
+| Command | Purpose |
+| --- | --- |
+| `./build/sbt core/assembly` | Builds the uber-jar required for Python tests. |
+| `./build/sbt doc` | Generates Scala API documentation. |
+| `./build/sbt +package` | Cross-builds artifacts for all supported Scala versions. |
+
+---
+
+## 4. Python workflow
+
+The Python package resides under `python/` and uses Poetry for dependency
+management.
+
+1. Build the GraphFrames assembly jar:
+   ```bash
+   ./build/sbt core/assembly
+   ```
+2. Capture the Scala version sbt selected (Spark 3.5.x defaults to 2.12.20):
+   ```bash
+   ./build/sbt "show core/scalaVersion"
+   export SCALA_VERSION=2.12.20   # adjust to the reported value
+   ```
+3. Install Python dependencies:
+   ```bash
+   cd python
+   poetry install --with dev
+   ```
+4. Run the Python test suite (requires `SPARK_HOME` and `SCALA_VERSION`):
+   ```bash
+   export SPARK_HOME=${SPARK_HOME:-$HOME/.local/spark/spark-3.5.6-bin-hadoop3}
+   ./run-tests.sh
```

Comment:
I'm afraid, that `./run-tests.sh` is very-very outdated... At the moment all the logic, related to the adding a correct JAR file to the testing SparkSession is in `conftest.py`: https://github.com/graphframes/graphframes/blob/main/python/tests/conftest.py

Long story short, it is much easier to run `dev/build_jar.py` and after that just `pytest -vvv`.

You can see how it is run in CI: https://github.com/graphframes/graphframes/blob/main/.github/workflows/python-ci.yml